compiler/rustc_data_structures/src/dyn_send_sync_impl.rs

```diff
@@ -80,34 +102,51 @@ cfg_if!(
 
         macro_rules! impls_dyn_sync {
             ($([$t2: ty $(where $($generics2: tt)*)?])*) => {
-                $(unsafe impl$(<$($generics2)*>)? DynSync for $t2 where $t2: Sync{})*
+                $(unsafe impl$(<$($generics2)*>)? DynSync for $t2 {})*
             };
         }
 
-        // Data structures which already impl `Sync`.
+        unsafe impl<T: ?Sized + DynSync + DynSend> DynSend for std::sync::Arc<T> {}
+        unsafe impl<T: ?Sized + DynSync + DynSend> DynSync for std::sync::Arc<T> {}
+
+        unsafe impl<T: DynSend + ?Sized> DynSend for std::ptr::Unique<T> {}
+        unsafe impl<T: DynSync + ?Sized> DynSync for std::ptr::Unique<T> {}
```

Comment:
These impls seem questionable.

Reply:
I use this since `RawVec` and `Box` use `Unique` inside. So we can impl for `Vec` and `Box` to avoid introduce `Unique`